outputs/amqp/amqp_test.go

```diff
@@ -0,0 +1,28 @@
+package amqp
+
+import (
+	"testing"
+
+	"github.com/influxdb/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConnectAndWrite(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	var url = "amqp://" + testutil.GetLocalHost() + ":5672/"
```

Comment:
It looks like you are following the form of the kafka output plugin here. In the case of kafka, we've added a kafka broker to the `scripts/docker-compose.yml` file that the test is connecting to in long mode. I'm not very familiar with AMQP, but is there a similar machine that you could add to the dockerfile? Would a rabbitmq docker container work? I think something like this should work:
```
rabbitmq:
    image: rabbitmq:3-management
    hostname: docker_rabbit
    ports:
        - "15672:15672"
        - "5672:5672"
```